bin/inference/pycbc_inference_plot_posterior

```diff
@@ -77,11 +78,15 @@ samples = samples if isinstance(samples, list) else [samples]
 # if a z-arg is specified, load samples for it
 if opts.z_arg is not None:
     logging.info("Getting samples for colorbar")
+    z_arg = 'loglikelihood' if opts.z_arg == 'snr' else opts.z_arg
     zlbl = opts.z_arg_labels[opts.z_arg]
     zvals = []
     for fp in fps:
-        zsamples = fp.samples_from_cli(opts, parameters=opts.z_arg)
-        zvals.append(zsamples[opts.z_arg])
+        zsamples = fp.samples_from_cli(opts, parameters=z_arg)
+        if opts.z_arg == 'snr':
+            loglr = zsamples[z_arg] - zsamples.lognl
+            zsamples[z_arg] = conversions.snr_from_loglr(loglr)
+        zvals.append(zsamples[z_arg])
```

Comment:
Make the change suggested above, then remove these changes.

Reply:
that won't work, `read_samples`  doesn't know about `snr_from_loglikelihood`. simplifying here requires other changes in other modules, this is the quick easy way to fix the bug

Reply:
actually, `read_samples` does know about `snr_from_loglr`, but `read_raw_samples` does not know about the attributes, so it complains when requesting for `lognl`. There are `read_raw_samples` scattered all over the place: in the posterior module
https://github.com/gwastro/pycbc/blob/master/pycbc/inference/io/posterior.py#L35
in the base_mcmc module
https://github.com/gwastro/pycbc/blob/master/pycbc/inference/io/base_mcmc.py#L356
in the base_multitemper module
https://github.com/gwastro/pycbc/blob/master/pycbc/inference/io/base_multitemper.py#L162
...
And probably every time a new sampler is added, then a new `read_raw_samples` function will appear. I find it much cleaner to just do it this way, rather than having all the `read_raw_samples` functions return attributes (especially since to add the attribute to the samples recordarray, one would have to add the same constant value to each item of the array)

Reply:
Ah, I forgot that read raw samples will interpret that as a parameter it needs to get. Ok, this is fine then.